e2e/test/scenarios/dashboard-filters/dashboard-filters-query-stages.cy.spec.ts

```diff
@@ -0,0 +1,1298 @@
+          ["Review", REVIEWS_DATE_COLUMNS],
+        ]);
+        verifyNoDashcardMappingOptions(QUESTION_BASED_MODEL_INDEX);
+        verifyNoDashcardMappingOptions(MODEL_BASED_MODEL_INDEX);
+      }
+
+      function verifyTextMappingOptions() {
+        verifyDashcardMappingOptions(QUESTION_BASED_QUESTION_INDEX, [
+          ["Summaries", ["Category", "Reviewer", "Category"]],
+          ["Review", REVIEWS_TEXT_COLUMNS],
+        ]);
+        verifyDashcardMappingOptions(MODEL_BASED_QUESTION_INDEX, [
+          ["Summaries", ["Category", "Reviewer", "Category"]],
+          ["Review", REVIEWS_TEXT_COLUMNS],
+        ]);
+        verifyDashcardMappingOptions(QUESTION_BASED_MODEL_INDEX, [
+          [
+            NAMELESS_SECTION,
+            [
+              "Reviews - Created At: Month → Reviewer",
+              "Products Via Product ID Category",
+            ],
+          ],
+        ]);
+        verifyDashcardMappingOptions(MODEL_BASED_MODEL_INDEX, [
+          [
+            NAMELESS_SECTION,
+            [
+              "Reviews - Created At: Month → Reviewer",
+              "Products Via Product ID Category",
+            ],
+          ],
+        ]);
+      }
+
+      function verifyNumberMappingOptions() {
+        verifyDashcardMappingOptions(QUESTION_BASED_QUESTION_INDEX, [
+          [
+            "Summaries",
+            ["Count", "Sum of Total", "5 * Count", "Count", "Sum of Rating"],
+          ],
+          ["Review", REVIEWS_NUMBER_COLUMNS],
+        ]);
+        verifyDashcardMappingOptions(MODEL_BASED_QUESTION_INDEX, [
+          [
+            "Summaries",
+            ["Count", "Sum of Total", "5 * Count", "Count", "Sum of Rating"],
+          ],
+          ["Review", REVIEWS_NUMBER_COLUMNS],
+        ]);
+        verifyNoDashcardMappingOptions(QUESTION_BASED_MODEL_INDEX);
+        verifyNoDashcardMappingOptions(MODEL_BASED_MODEL_INDEX);
+      }
+    });
+  });
+});
+
+function createBaseQuestions() {
+  createQuestion({
+    type: "question",
+    name: "Q0 Orders",
+    description: "Question based on a database table",
+    query: {
+      "source-table": ORDERS_ID,
+    },
+  }).then(response => cy.wrap(response.body).as("ordersQuestion"));
+
+  cy.then(function () {
+    createQuestion({
+      type: "question",
+      name: "Base Orders Question",
+      query: {
+        "source-table": `card__${this.ordersQuestion.id}`,
+      },
+    }).then(response => cy.wrap(response.body).as("baseQuestion"));
+
+    createQuestion({
+      type: "model",
+      name: "Base Orders Model",
+      query: {
+        "source-table": `card__${this.ordersQuestion.id}`,
+      },
+    }).then(response => cy.wrap(response.body).as("baseModel"));
+  });
+}
+
+// Q1 - join, custom column, no aggregations, no breakouts
+function createQ1Query(source: Card): StructuredQuery {
+  return {
+    "source-table": `card__${source.id}`,
+    expressions: {
+      Net: ["-", TOTAL_FIELD, TAX_FIELD],
+    },
+    joins: [
```

Comment:
We never use `metabase-lib` in e2e tests to create queries but maybe we should start :thinking: